Packages/com.unity.inputsystem/InputSystem/Controls/InputControlExtensions.cs

```diff
@@ -1583,6 +1582,8 @@ public static DeviceBuilder Setup(this InputDevice device, int controlCount, int
             device.m_Device = device;
 
             device.m_ChildrenForEachControl = new InputControl[controlCount];
+            device.m_UpdatedButtons = new HashSet<int>();
```

Comment:
Do you think it would be relevant to initialize the capacity of this hash-set based off the number of buttons available on the device? This should be fixed so its basically a "constant" based off the device descriptor.

Reply:
Code below here basically seems to count the number of buttons anyway